src/RevProc.cc

```diff
@@ -550,6 +550,7 @@ RevInst RevProc::DecodeCIInst(uint16_t Inst, unsigned Entry){
       // sign extend
       CompInst.imm |= 0b11111111111111000000000000000000;
     }
+    CompInst.imm >>= 12;  //immd value will be re-aligned on execution
```

Comment:
There is a difference in how c.lui and lui handle immd values - lui assumes they are aligned to the LSB position and because the implementation of c.lui calls lui this results in the immd in c.lui being left-shited twice. So compensating here in the decode by aligning the c.lui immd to the LSB position